test/node-rpc-test.js

```diff
@@ -0,0 +1,230 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+'use strict';
+
+const assert = require('bsert');
+const FullNode = require('../lib/node/fullnode');
+const {NodeClient, WalletClient} = require('hs-client');
+const KeyRing = require('../lib/primitives/keyring');
+const pkg = require('../lib/pkg');
+
+// will this work?
+const blake2b = require('bcrypto/lib/blake2b');
+const base64 = require('bcrypto/lib/encoding/base64');
+
+const MAGIC_STRING = `${pkg.currency} signed message:\n`;
+
+const ports = {
+  p2p: 49331,
+  node: 49332,
+  wallet: 49333
+};
+const node = new FullNode({
+  network: 'regtest',
+  apiKey: 'foo',
+  walletAuth: true,
+  memory: true,
+  workers: true,
+  workersSize: 2,
+  plugins: [require('../lib/wallet/plugin')],
+  port: ports.p2p,
+  httpPort: ports.node,
+  env: {
+    'HSD_WALLET_HTTP_PORT': ports.wallet.toString()
+  }
+});
+
+const nclient = new NodeClient({
+  port: ports.node,
+  apiKey: 'foo',
+  timeout: 15000
+});
+
+const wclient = new WalletClient({
+  port: ports.wallet,
+  apiKey: 'foo'
+});
+
+describe('RPC', function() {
+  this.timeout(15000);
+
+  before(async () => {
+    await node.open();
+    await nclient.open();
+  });
+
+  after(async () => {
+    await nclient.close();
+    await node.close();
+  });
+
+  it('should rpc help', async () => {
+    assert(await nclient.execute('help', []));
+    await assert.rejects(async () => {
+      await nclient.execute('help', ['getinfo']);
+    }, {
+      name: 'Error',
+      message: /^getinfo/
+    });
+  });
+  it('should rpc getinfo', async () => {
+    const info = await nclient.execute('getinfo', []);
+    assert.strictEqual(info.blocks, 0);
+  });
+
+  describe('getblock', function () {
+    it('should rpc getblock', async () => {
+      const {chain} = await nclient.getInfo();
+      const info = await nclient.execute('getblock', [chain.tip]);
+
+      const properties = [
+        'hash', 'confirmations', 'strippedsize',
+        'size', 'weight', 'height', 'version',
+        'versionHex', 'merkleroot', 'coinbase',
+        'tx', 'time', 'mediantime', 'nonce',
+        'bits', 'difficulty', 'chainwork',
+        'nTx', 'previousblockhash', 'nextblockhash'
+      ];
```

Comment:
could also include some of the hsd-exclusive stuff `witnessRoot`, `treeRoot`, etc, and compare against the `chain.tip` entry in the assertion tests below